### PR TITLE
Revert "[SPARK-35028][SQL] ANSI mode: disallow group by aliases"

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Revert [[SPARK-35028][SQL] ANSI mode: disallow group by aliases ](https://github.com/apache/spark/pull/32129)

### Why are the changes needed?

It turns out that many users are using the group by alias feature.  Spark has its precedence rule when alias names conflict with column names in Group by clause: always use the table column. This should be reasonable and acceptable.
Also, external DBMS such as PostgreSQL and MySQL allow grouping by alias, too. 

As we are going to announce ANSI mode GA in Spark 3.2, I suggest allowing the group by alias in ANSI mode.

### Does this PR introduce _any_ user-facing change?

No, the feature is not released yet.

### How was this patch tested?

Unit tests

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/inputs/ansi/group-analytics.sql [hunk 1]
```diff
@@ -1 +0,0 @@
---IMPORT group-analytics.sql
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/ansi/group-analytics.sql:1]: Do we need to remove the result file?

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/ansi/group-analytics.sql:1]: good catch!

Comment (gengliangwang) [reply sql/core/src/test/resources/sql-tests/inputs/ansi/group-analytics.sql:1]: Thanks, removed

**Comment by gengliangwang** [comment]
Merging to master/3.2

**Comment by dongjoon-hyun** [comment]
+1, LGTM.

**Pull request closed without merging**